pkg/manifests/manifests_test.go

```diff
@@ -3315,6 +3325,102 @@ func TestTelemeterConfiguration(t *testing.T) {
 	}
 }
 
+func TestTelemeterClientSecret(t *testing.T) {
+	for _, tc := range []struct {
+		name         string
+		config       string
+		existingData map[string][]byte
+		expectedData map[string][]byte
+		tokenChanged bool
+	}{
+		{
+			name: "No existing secret",
+			config: `telemeterClient:
+  token: mySecretToken
+`,
+			existingData: map[string][]byte{},
+			expectedData: map[string][]byte{
+				"token": []byte("mySecretToken"),
+			},
+			tokenChanged: true,
+		},
+		{
+			name: "Existing secret",
+			config: `telemeterClient:
+  token: mySecretToken
+`,
+			existingData: map[string][]byte{
+				"token": []byte("mySecretToken"),
+				"salt":  []byte("1234456789ABCDEF"),
+			},
+			expectedData: map[string][]byte{
+				"token": []byte("mySecretToken"),
+				"salt":  []byte("1234456789ABCDEF"),
+			},
+			tokenChanged: false,
+		},
+		{
+			name: "Existing secret, salt gets deleted",
+			config: `telemeterClient:
+  token: mySecretToken
+`,
+			existingData: map[string][]byte{
+				"token": []byte("mySecretToken"),
+			},
+			expectedData: map[string][]byte{
+				"token": []byte("mySecretToken"),
+			},
+			tokenChanged: true,
```

Comment:
Nit: I understand you need this field because the salt is randomly generated so you cannot add the expected value of the salt in the test data. But it is a bit misleading having `tokenChanged = true` here because the token hasn't changed in this case. Renaming `tokenChanged` to something like `saltUpdateExpected` would be more readable